### ISSUE TITLE
Upgrade to version  1.9.1 causes segmentation fault and shop not working anymore

### ISSUE THREAD

**Issue by richardreen** [tig-nl/postnl-magento2#156]
Tried to upgrade to version 1.9.1, we are using now version 1.8.1
During upgrade some message that we have to reinstall this extension and after that the setup:upgrade always fails with a segmentation fault. No error messages in the log.

Here is hole logging of the upgrade.

[uxxxxxx@xxxxx public_html]$ bin/magento maintenance:enable
Enabled maintenance mode
[uxxxxxx@sxxxx public_html]$ composer update tig/postnl-magento2
Loading composer repositories with package information
Updating dependencies (including require-dev)                                                
Package operations: 0 installs, 1 update, 0 removals
  - Updating tig/postnl-magento2 (1.8.1 => 1.9.0):  Checking out d03a2a15fc
    Update failed (Failed to execute git checkout 'd03a2a15fcfbc553c857732b0ebcf8c95234e86a' -- && git reset --hard 'd03a2a15fcfbc553c857732b0ebcf8c95234e86a' --

error: The following untracked working tree files would be overwritten by checkout:
        Config/CheckoutConfiguration/CheckoutExtension.php
        Config/Provider/ReturnOptions.php
        Plugin/Order/AsyncPlugin.php
        Plugin/Postcodecheck/Mageplaza/LayoutProcessor.php
        Service/Shipment/Packingslip/Compatibility/DataHelperFactoryProxy.php
        Service/Shipment/Packingslip/Compatibility/GeneratePdfFactoryProxy.php
        Service/Shipment/Packingslip/Compatibility/XtentoPdfCustomizer.php
        Setup/V182/Schema/ModifyConditionNameColumn.php
        Setup/V190/Schema/InstallReturnBarcode.php
        Setup/V190/Schema/InstallReturnLabel.php
        Test/Unit/Service/Shipment/Label/Type/DomesticTest.php
        etc/adminhtml/system/postnlsettings/returns-be.xml
        etc/adminhtml/system/postnlsettings/returns-nl.xml
Please move or remove them before you can switch branches.
Aborting

    Would you like to try reinstalling the package instead [yes]? 
  - Removing tig/postnl-magento2 (1.8.1)
  - Installing tig/postnl-magento2 (1.9.0): Cloning d03a2a15fc
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package facebook/webdriver is abandoned, you should avoid using it. Use php-webdriver/webdriver instead.
Package guzzlehttp/ringphp is abandoned, you should avoid using it. No replacement was suggested.
Package guzzlehttp/streams is abandoned, you should avoid using it. No replacement was suggested.
Package leafo/scssphp is abandoned, you should avoid using it. Use scssphp/scssphp instead.
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Package zendframework/zend-barcode is abandoned, you should avoid using it. Use laminas/laminas-barcode instead.
Package zendframework/zend-captcha is abandoned, you should avoid using it. Use laminas/laminas-captcha instead.
Package zendframework/zend-code is abandoned, you should avoid using it. Use laminas/laminas-code instead.
Package zendframework/zend-config is abandoned, you should avoid using it. Use laminas/laminas-config instead.
Package zendframework/zend-console is abandoned, you should avoid using it. Use laminas/laminas-console instead.
Package zendframework/zend-crypt is abandoned, you should avoid using it. Use laminas/laminas-crypt instead.
Package zendframework/zend-db is abandoned, you should avoid using it. Use laminas/laminas-db instead.
Package zendframework/zend-di is abandoned, you should avoid using it. Use laminas/laminas-di instead.
Package zendframework/zend-diactoros is abandoned, you should avoid using it. Use laminas/laminas-diactoros instead.
Package zendframework/zend-dom is abandoned, you should avoid using it. Use laminas/laminas-dom instead.
Package zendframework/zend-escaper is abandoned, you should avoid using it. Use laminas/laminas-escaper instead.
Package zendframework/zend-eventmanager is abandoned, you should avoid using it. Use laminas/laminas-eventmanager instead.
Package zendframework/zend-feed is abandoned, you should avoid using it. Use laminas/laminas-feed instead.
Package zendframework/zend-filter is abandoned, you should avoid using it. Use laminas/laminas-filter instead.
Package zendframework/zend-form is abandoned, you should avoid using it. Use laminas/laminas-form instead.
Package zendframework/zend-http is abandoned, you should avoid using it. Use laminas/laminas-http instead.
Package zendframework/zend-hydrator is abandoned, you should avoid using it. Use laminas/laminas-hydrator instead.
Package zendframework/zend-i18n is abandoned, you should avoid using it. Use laminas/laminas-i18n instead.
Package zendframework/zend-inputfilter is abandoned, you should avoid using it. Use laminas/laminas-inputfilter instead.
Package zendframework/zend-json is abandoned, you should avoid using it. Use laminas/laminas-json instead.
Package zendframework/zend-loader is abandoned, you should avoid using it. Use laminas/laminas-loader instead.
Package zendframework/zend-log is abandoned, you should avoid using it. Use laminas/laminas-log instead.
Package zendframework/zend-mail is abandoned, you should avoid using it. Use laminas/laminas-mail instead.
Package zendframework/zend-math is abandoned, you should avoid using it. Use laminas/laminas-math instead.
Package zendframework/zend-mime is abandoned, you should avoid using it. Use laminas/laminas-mime instead.
Package zendframework/zend-modulemanager is abandoned, you should avoid using it. Use laminas/laminas-modulemanager instead.
Package zendframework/zend-mvc is abandoned, you should avoid using it. Use laminas/laminas-mvc instead.
Package zendframework/zend-psr7bridge is abandoned, you should avoid using it. Use laminas/laminas-psr7bridge instead.
Package zendframework/zend-serializer is abandoned, you should avoid using it. Use laminas/laminas-serializer instead.
Package zendframework/zend-server is abandoned, you should avoid using it. Use laminas/laminas-server instead.
Package zendframework/zend-servicemanager is abandoned, you should avoid using it. Use laminas/laminas-servicemanager instead.
Package zendframework/zend-session is abandoned, you should avoid using it. Use laminas/laminas-session instead.
Package zendframework/zend-soap is abandoned, you should avoid using it. Use laminas/laminas-soap instead.
Package zendframework/zend-stdlib is abandoned, you should avoid using it. Use laminas/laminas-stdlib instead.
Package zendframework/zend-text is abandoned, you should avoid using it. Use laminas/laminas-text instead.
Package zendframework/zend-uri is abandoned, you should avoid using it. Use laminas/laminas-uri instead.
Package zendframework/zend-validator is abandoned, you should avoid using it. Use laminas/laminas-validator instead.
Package zendframework/zend-view is abandoned, you should avoid using it. Use laminas/laminas-view instead.
Writing lock file
Generating autoload files

Generating autoload files
[uxxxxxx@xxxxxx public_html]$ bin/magento setup:upgrade
Segmentatiefout

[uxxxxxx@xxxxxx public_html]$ bin/magento -vvv setup:upgrade
Segmentatiefout

** Please complete the following information**
- Magento version: [e.g. Magento 2.3.3
- PHP version: [e.g. 7.3.13]
- TIG PostNL version: [e.g. 1.8.1]

NB. We have another site on the same server using excactly the same resources but with the tig/postnl extension disabled. Here are no issues if we de an upgrade to for example magento 2.3.4. 
This site can not be upgraded because the tig/postnl module is also upgraded and is causing the segmentation fault.

**Comment by tig-jaspersmits** [comment]
Hi @richardreen ,

Thanks for your message. I'll discuss this with my colleague and get back to you with a response. @tig-dennisvanderhammen , any thoughts on this problem?

Kind regards,
Jasper
TIG

**Comment by tig-dennisvanderhammen** [comment]
Hi @richardreen,

The first messages that you are receiving are caused because Composer detected changes in vendor/tig/postnl-magento2. It could be that this is caused by a specific server command or intentional changes. We always recommend not changing files in the vendor folder.

I would also recommend - if this isn't the case yet - to add toe vendor map to the .gitignore file of your repository.

It seems that Composer has a workaround for this already, which is removing PostNL Magento 2 version v1.8.1 and installing version v1.9.0.

The error "Segmentatiefout" seems to be an error that isn't typical of PHP. I would assume something in the server system is causing issues. I would recommend checking your system logs or nginx/apache logs to see if there are any errors. If you can't find anything, I would recommend getting in touch with your hosting provider, as this could also be caused by hardware issues.

Kind regards,
Dennis
TIG

**Comment by richardreen** [comment]
We did not change any file in the vendor tig directories. As you could also read we have another site that is using another directory with excaclty the same content as for directory/shop but one difference...the tig/postnl for magento2 is disabled and in the directory we can ugrade without any problems....so this problem s is still there and we can not upgrade...why is this ticked closed now?????

**Comment by tig-dennisvanderhammen** [comment]
Hi Richard,

About your first issue, the changed files, this is something I could ascertain because of the error messages you have provided. The message _"error: The following untracked working tree files would be overwritten by checkout" is something"_ is an error that Composer throws when the files in the vendor map are not equal to the version on packagist.org. This could for example be changes in the permissions, and as I stated in my previous message, could be caused by a server command.

Because Composer removed version v1.8.1 and reinstalled version v1.9.0, this should not cause any issues.

As for your second issue, the setup:upgrade part: 
We use our Git repositories' issue system for problems that are caused by the codebase. I have closed your issue because we are unable to reproduce your problem on a core Magento environment. If you are unable to find any logs in your Magento environment or server environment that could indicate any faults, I would recommend getting in touch with PostNL's servicedesk by calling them on 088-2255651 or by sending them an email at citservicedesk@postnl.nl.
